scripts/Muon/GUI/Common/corrections_tab_widget/background_corrections_model.py

```diff
@@ -320,8 +325,23 @@ def get_counts_workspace_name(self, run_string: str, group: str, rebin: bool) ->
 
     def default_x_range(self, run: str, group: str, rebin: bool) -> tuple:
         """Returns the x range to use by default for the background corrections. It is the second half of the data."""
+        run_list = run_string_to_list(run)
+        last_good_data = self._context.last_good_data(run_list)
+
+        if last_good_data != 0.0:
+            return self._get_x_range_from(self._context.first_good_data(run_list), last_good_data)
+        else:
+            return self._get_x_range_from_counts_workspace(run, group, rebin)
+
+    @staticmethod
+    def _get_x_range_from(first_good_data: float, last_good_data: float) -> tuple:
+        """Get the default start and end X from the first and last good data, and the time zero."""
+        return (last_good_data - first_good_data) / 2.0 + first_good_data, last_good_data
```

Comment:
could we make this a function since both this and the one below do the same calculation? Then in future if we need to change it, we only need to do it once